src/components/organisms/AdminVenueView/AdminVenueView.tsx

```diff
@@ -36,22 +44,25 @@ export const AdminVenueView: React.FC = () => {
   const { isCurrentVenueLoaded } = useConnectCurrentVenueNG(venueId);
 
   const renderAdminVenueTabs = useMemo(() => {
-    return Object.entries(adminVenueTabLabelMap).map(([key, text]) => (
-      <Nav.Link
+    return Object.entries(adminVenueTabLabelMap).map(([key, label]) => (
+      <Link
         key={key}
-        className={classNames("AdminVenueView__tab", {
+        to={adminNGVenueUrl(venueId, key)}
+        className={classNames({
+          AdminVenueView__tab: true,
           "AdminVenueView__tab--selected": selectedTab === key,
         })}
-        eventKey={key}
       >
-        {text}
-      </Nav.Link>
+        {label}
+      </Link>
     ));
-  }, [selectedTab]);
+  }, [selectedTab, venueId]);
 
-  const selectTab = useCallback((tab: string) => {
-    setSelectedTab(tab as AdminVenueTab);
-  }, []);
+  useEffect(() => {
+    if (venueId) return;
+    // when URL is invalid, land on the root one for the admin
+    history.push(adminNGRootUrl());
```

Comment:
How is this going to work, if my internet is slow and the venueId is undefined until, the request is complete. This should be somehow handled from the router dep

Reply:
The pains of hooks made me move it before the 
```ts
  if (!isCurrentVenueLoaded) {
    return <LoadingPage />;
  }
```

which should still execute before any `useEffect()` does, as effects scheduled after rendering.

But still, if `venueId` is undefined, that's since the start of the request.
That value is retrieved by `useParams()` and if `undefined`,  no loading will ever start in the first place.

Reply:
This kind of logic could potentially be pulled up to the router level, but that's just putting internal children logic into the parent. Could still work with some generic router handling though, one that doesn't account for children specifics.

Reply:
I think you should talk to product about these things before implementing them, have you or did you add this as your own personal decision/preference? I don't think it's the wrong thing to do, but it might be confusing for someone to copy/paste a link and then get redirected to `/admin-ng/`. We also had this venue doesn't exist page but i'm not sure if that was for the template wrapper only.

Reply:
Removed the automatic re-direct as OOS @ 2c33b6b8de126d7a4550fdd2ca5b20e1ad2b5168 and will create separate PR once @sparkletown/product is being consulted.